adapters/eplanning/eplanning.go

```diff
@@ -0,0 +1,147 @@
+package eplanning
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type EPlanningAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+type EPlanningRequest struct {
+	id      string
+	user    *openrtb.User
+	device  *openrtb.Device
+	adUnits []*EPlanningAdUnit
+}
+
+type EPlanningBid struct {
+	Id     string  `json:"id"`
+	Price  float64 `json:"price,omitempty"`
+	Width  uint64  `json:"w,omitempty"`
+	Height uint64  `json:"h,omitempty"`
+	DealId string  `json:"dealid,omitempty"`
+}
+
+type EPlanningAdUnit struct {
+	Id         string
+	Bidfloor   float64
+	Instl      int8
+	SspSpaceId int `json:"ssp_espacio_id,omitempty"`
+	Video      *openrtb.Video
+	Banner     *openrtb.Banner
+}
+
+func (adapter *EPlanningAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	ePlanningRequest, errors := openRtbToEPlanningRequest(request)
+	if len(ePlanningRequest.adUnits) == 0 {
+		return nil, errors
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	requestData := adapters.RequestData{
+		Method: "POST",
+		Uri:    adapter.URI,
+		Body:   reqJSON,
+	}
+
+	requests := []*adapters.RequestData{&requestData}
+
+	return requests, errors
+}
+
+func openRtbToEPlanningRequest(request *openrtb.BidRequest) (*EPlanningRequest, []error) {
+
+	adUnits := make([]*EPlanningAdUnit, 0, len(request.Imp))
+	errors := make([]error, 0, len(request.Imp))
+	for _, imp := range request.Imp {
+		var params openrtb_ext.ExtImpEPlanning
+		err := json.Unmarshal(imp.Ext, &params)
```

Comment:
You are skipping a layer. You will want to first unmarshal the `bidder` element of `imp.Ext`, and then unmarshal your parameters from that.

```
                var bidderExt adapters.ExtImpBidder
	        if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
                    errors = append(errors, err)
		    return nil, errors
	        }

      		var params openrtb_ext.ExtImpEPlanning
		err := json.Unmarshal(bidderExt.Bidder, &params)
		if err != nil {
			errors = append(errors, err)
			continue
		}
```